### PR TITLE
DOCSP-37537 Delete Operations

### PR DESCRIPTION
**Pull request by jordan-smith721**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-37537
Staging - https://preview-mongodbjordansmith721.gatsbyjs.io/pymongo/DOCSP-37537-delete/write/delete/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/write/delete.txt [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+..    top of the page. To expand the lab to a full-screen format, click the
+..    full-screen button (:guilabel:`⛶`) in the top-right corner of the lab pane.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<pymongo-get-started>` tutorial.
```
Comment (jordan-smith721) [inline source/write/delete.txt:46]: Broken link until Get Started PR is merged

File: source/write/delete.txt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+Delete Operations
+-----------------
+
+You can perform delete operations in MongoDB by using the following methods:
+
+- ``delete_one()``, which deletes *the first document* that matches the search criteria
+- ``delete_many()``, which deletes *all documents* that match the search criteria
+
+Each delete method requires a **query filter** document, which specifies the
+search criteria that determines which documents to select for removal.
```
Comment (mongoKart) [inline source/write/delete.txt:57]:
```suggestion
search criteria that determine which documents to select for removal.
```

File: source/write/delete.txt [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+   * - Property
+     - Description
+
+   * - ``collation``
+     - | Specifies the kind of language collation to use when sorting
+         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`
+         in the MongoDB Server manual.
+
+   * - ``hint``
+     - | Gets or sets the index used to scan for documents.
```
Comment (mongoKart) [inline source/write/delete.txt:107]:
```suggestion
     - | Gets or sets the index to scan for documents. 
```

File: source/write/delete.txt [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+   * - ``hint``
+     - | Gets or sets the index used to scan for documents. 
+         For more information, see the :manual:`hint statement </reference/command/delete/#std-label-deletes-array-hint>`
+         in the MongoDB Server manual.
+
+   * - ``session``
+     - | An instance of ``ClientSession``.
+
+   * - ``let``
+     - | A Map of parameter names and values. Values must be constant or closed
```
Comment (mongoKart) [inline source/write/delete.txt:115]: If Map is a data type, might be good to monospace

```suggestion
     - | A ``Map`` of parameter names and values. Values must be constant or closed
```

File: source/write/delete.txt [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+         MongoDB Server manual.
+
+   * - ``comment``
+     - | A comment to attach to the operation. For more information, see the :manual:`delete command
+         fields </reference/command/delete/#command-fields>` guide in the
+         MongoDB Server manual for more information.
+
+The following code uses the ``delete_many()`` method to delete all documents in
+the ``restaurants`` collection with a ``name`` value that includes the string ``"Mongo"``.
+It uses the ``comment`` option to add a comment to the operation:
```
Comment (mongoKart) [inline source/write/delete.txt:128]:
```suggestion
It also uses the ``comment`` option to add a comment to the operation:
```

File: source/write/delete.txt [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+Return Value
+~~~~~~~~~~~~
+
+The ``delete_one()`` and ``delete_many()`` methods return a
+``DeleteResult`` type. This type contains the following properties:
+
+- ``deleted_count``, which indicates the number of documents deleted.
+- ``acknowledged``, which indicates if the server acknowledges the result.
+- ``raw_result``, which is the raw result returned by the server.
```
Comment (mongoKart) [inline source/write/delete.txt:150]: These are sentence fragments, so no periods needed

```suggestion
- ``deleted_count``, which indicates the number of documents deleted
- ``acknowledged``, which indicates if the server acknowledges the result
- ``raw_result``, which is the raw result returned by the server
```

File: source/write/delete.txt [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+.. note::
+
+   If the ``acknowledged`` attribute is ``False``, all other attributes of ``DeleteResult``
+   raise an ``InvalidOperation`` exception when accessed. The driver cannot
+   determine these values if the server does not acknowledge the write operation.
+
+If the query filter does not match any documents, the driver doesn't delete any
+documents and ``deleted_count`` is 0.
+
+API Documentation
```
Comment (mongoKart) [inline source/write/delete.txt:161]: Consider including `ClientSession` here.

**Review by mongoKart — APPROVED** [review]
lgtm! really nicely done. just a few small suggestions.

File: source/write/delete.txt (hunk 4 above)
Comment (jordan-smith721) [reply source/write/delete.txt]: Just a typo

**Pull request merged**